felix/ipsets/utils_for_test.go

```diff
@@ -108,10 +108,12 @@ func (d *mockDataplane) newCmd(name string, arg ...string) CmdIface {
 			SetName:   name,
 		}
 	case "list":
-		Expect(len(arg)).To(Equal(1))
+		Expect(len(arg)).To(Equal(2))
 		cmd = &listCmd{
 			Dataplane: d,
 			resultC:   make(chan error),
+			allIpSets: arg[1] == "-name",
+			SetName:   arg[1], // either ipset name or '-name' to return the list of all ipsets
```

Comment:
I'd prefer you to set this to "" if -name is specified.  OK to only handle some cases but watch out for setting fields to misleading values that might get picked up later on.